pkg/storage/image.go

```diff
@@ -284,11 +328,26 @@ func GetImageService(store storage.Store, defaultTransport string, insecureRegis
 		}
 	}
 
+	seenRegistries := make(map[string]bool, len(registries))
+	cleanRegistries := []string{}
+	for _, r := range registries {
+		if seenRegistries[r] {
+			continue
+		}
+		cleanRegistries = append(cleanRegistries, r)
+		seenRegistries[r] = true
+	}
+
+	if len(cleanRegistries) == 0 {
+		return nil, errors.New("no registries configured. Please add registries with the --registry flag")
```

Comment:
Why am I required to have a registry defined?  Can't a user say that he always requires fully qualified names?